src/store/singing.ts

```diff
@@ -3213,6 +3215,171 @@ export const singingStore = createPartialStore<SingingStoreTypes>({
       return Math.max(1, lastNoteEndTime + 1);
     },
   },
+
+  EXPORT_SONG_PROJECT: {
+    action: createUILockAction(
+      async (
+        { state, getters, actions },
+        { fileType, fileTypeLabel },
+      ): Promise<SaveResultObject> => {
+        const fileBaseName = generateDefaultSongFileBaseName(
+          getters.PROJECT_NAME,
+          getters.SELECTED_TRACK,
+          getters.CHARACTER_INFO,
+        );
+        const project = ufProjectFromVoicevox(
+          {
+            tempos: state.tempos,
+            timeSignatures: state.timeSignatures,
+            tpqn: state.tpqn,
+            tracks: state.trackOrder.map((trackId) =>
+              getOrThrow(state.tracks, trackId),
+            ),
+          },
+          fileBaseName,
+        );
+
+        // 複数トラックかつ複数ファイルの形式はディレクトリを選択する
+        if (
+          state.trackOrder.length > 1 &&
+          ["ust", "xml", "musicxml"].includes(fileType)
+        ) {
+          const dirPath = await window.backend.showSaveDirectoryDialog({
+            title: "プロジェクトを書き出し",
+          });
+          if (!dirPath) {
+            return {
+              result: "CANCELED",
+              path: "",
+            };
+          }
```

Comment:
（細かいけど）
こういう小さいの、１行にするとだいぶ見やすくなるので個人的におすすめです。
```suggestion
          if (!dirPath) {
            return { result: "CANCELED", path: "" };
          }
```

ちなみに最初の改行を消してフォーマットかけると１行になります。
```
{
  "hoge":
```
↓
```
{"hoge":
```
↓
フォーマット

見た目は慣れ！